lighthouse-core/gather/gather-runner.js

```diff
@@ -271,7 +278,9 @@ class GatherRunner {
     const networkRecords = NetworkRecorder.recordsFromLogs(devtoolsLog);
     log.verbose('statusEnd', status);
 
-    let pageLoadError = GatherRunner.getPageLoadError(passContext.url, networkRecords);
+    const securityState = await driver.getSecurityState();
```

Comment:
Maybe we're making this harder than it needs to be by trying to unify with with the other page load errors.

`getPageLoadError` is trying really hard to not interrupt normal control flow, inserting the promise rejection into the gather results instead of into the executing chain. Meanwhile, this security check is trying to abandon the gathering altogether.

Maybe this should just be a separate function, `checkPageSecurityState()` or whatever, called here-ish. If the page is insecure, throw in there. It'll bail back to here, which will bail up to the `catch` in `run()`. If we really want to still return something in that case, the check against `LHError.errors.INSECURE_DOCUMENT_REQUEST.code` can happen there (or rethrow if it's something different).

This also gets around extending `passData`, which makes sense, because really there isn't any other useful pass data when there's this kind of error.

WDYT?

Reply:
Can't quite figure out how to still display a LHR in `run`'s `catch`. I marked the security LHError fatal (so it would throw), moved `baseArtifacts` up a bit (so it can be accessed in `catch`), but get this silly error:

```
Error: CSSUsage failed to provide an artifact.
    at Function.collectArtifacts (/Users/cjamcl/src/lighthouse/lighthouse-core/gather/gather-runner.js:366:15)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
```